hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/SyncFuture.java

```diff
@@ -65,16 +74,30 @@
    */
   private Throwable throwable;
 
-  private Thread t;
+  /*
+   * Fields below are created once at reset() and accessed without any lock. Should be ok as they
+   * are immutable for this instance of sync future until it is reset.
+   */
+
+  /**
+   * The transaction id of this operation, monotonically increases.
+   */
+  private long txid;
 
   private boolean forceSync;
 
+  SyncFuture() {
+    this.doneLock = new ReentrantLock();
```

Comment:
These two fields could be marked as final if we never change them after construction.